window/src/os/x11/connection.rs

```diff
@@ -239,12 +239,22 @@ impl ConnectionOps for XConnection {
             .ok_or_else(|| anyhow::anyhow!("no screens were found"))?
             .clone();
 
-        // We don't yet know how to determine the active screen,
-        // so assume the main screen.
-        // TODO: find focused window and resolve it!
-        // Maybe something like <https://stackoverflow.com/a/43666928/149111>
-        // but ported to Rust?
-        let active = main.clone();
+        self.conn.send_request(&xcb::x::GrabServer {});
```

Comment:
I was going to suggest that we need to ensure that we ungrab the server in all of the error paths below (each `?` is potentially an early return!), to avoid locking things up (see: https://tronche.com/gui/x/xlib/window-and-session-manager/XGrabServer.html), but I don't think we need to grab it at all here.

In the stackoverflow link someone suggested grabbing the server while walking the tree of windows to prevent the focus changing out from under the client, but I don't think that matters here: we're only looking at a single window, so grabbing just means the user's focus change happens a fraction of a second later.

So let's remove the grab/ungrab logic to streamline and simplify a little.

```suggestion
```